plugins/connection/nsenter.py

```diff
@@ -0,0 +1,250 @@
+# Based on Ansible local connection plugin by:
+# (c) 2012 Michael DeHaan <michael.dehaan@gmail.com>
+# (c) 2015, 2017 Toshio Kuratomi <tkuratomi@ansible.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: nsenter
+    short_description: execute on host running controller container
+    description:
+        - This connection plugin allows Ansible, running in a privileged container, to execute tasks on the container
+          host instead of in the container itself.
+        - This is useful for running Ansible in a pull model, while still keeping the Ansible control node
+          containerized.
+        - It relies on having privileged access to run nsenter in the host's PID namespace, allowing it to enter the
+          namespace of PID 1.
+    author: Jeff Goldschrafe (@jgoldschrafe)
+    options:
+        host_volume_mount:
+            description:
+                - Host volume mount to read and write host files through
+            default: /host
+            vars:
+                - name: ansible_host_volume_mount
+            env:
+                - name: ANSIBLE_HOST_VOLUME_MOUNT
+            ini:
+                - section: nsenter_connection
+                  key: host_volume_mount
+        nsenter_pid:
+            description:
+                - PID to attach with using nsenter.
+                - The default should be fine unless you're attaching as a non-root user.
+            default: '1'
+            vars:
+                - name: ansible_nsenter_pid
+            env:
+                - name: ANSIBLE_NSENTER_PID
+            ini:
+                - section: nsenter_connection
+                  key: nsenter_pid
+        pipelining:
+            default: ANSIBLE_PIPELINING
+            description:
+                - Pipelining reduces the number of connection operations required to execute a module on the remote
+                  server, by executing many Ansible modules without actual file transfers.
+                - This can result in a very significant performance improvement when enabled.
+                - However this can conflict with privilege escalation (become).
+                  For example, when using sudo operations you must first disable 'requiretty' in the sudoers file for
+                  the target hosts, which is why this feature is disabled by default.
+            env:
+                - name: ANSIBLE_PIPELINING
+            ini:
+                - section: defaults
+                  key: pipelining
+            type: boolean
+            vars:
+                - name: ansible_pipelining
+    notes:
+        - The remote user is ignored; this plugin always runs as root.
+        - This plugin requires the container to be launched in the following way:
+            - The container image contains the C(nsenter) program
```

Comment:
Having an enumeration inside a notes paragraph doesn't work (i.e. it will look horribly). Try to format this as one paragraph, like "Bla (1) foo, (2) bar, (3) baz."